BitwardenShared/UI/Auth/StartRegistration/StartRegistrationViewTests.swift

```diff
@@ -0,0 +1,112 @@
+    var processor: MockProcessor<StartRegistrationState, StartRegistrationAction, StartRegistrationEffect>!
+    var subject: StartRegistrationView!
+
+    // MARK: Setup & Teardown
+
+    override func setUp() {
+        super.setUp()
+        processor = MockProcessor(state: StartRegistrationState())
+        let store = Store(processor: processor)
+        subject = StartRegistrationView(store: store)
+    }
+
+    override func tearDown() {
+        super.tearDown()
+        processor = nil
+        subject = nil
+    }
+
+    // MARK: Tests
+
+    /// Tapping the cancel button dispatches the `.dismiss` action.
+    func test_cancelButton_tap() throws {
+        let button = try subject.inspect().find(button: Localizations.cancel)
+        try button.tap()
+        XCTAssertEqual(processor.dispatchedActions.last, .dismiss)
+    }
+
+    /// Updating the text field dispatches the `.emailTextChanged()` action.
+    func test_emailField_updateValue() throws {
+        let textfield = try subject.inspect().find(viewWithId: Localizations.emailAddress).textField()
+        try textfield.setInput("text")
+        XCTAssertEqual(processor.dispatchedActions.last, .emailTextChanged("text"))
+    }
+
+    /// Updating the text field dispatches the `.nameTextChanged()` action.
+    func test_nameField_updateValue() throws {
+        let textfield = try subject.inspect().find(viewWithId: Localizations.name).textField()
+        try textfield.setInput("user name")
+        XCTAssertEqual(processor.dispatchedActions.last, .nameTextChanged("user name"))
+    }
+
+    /// Tapping the continue button performs the `.StartRegistration` effect.
+    func test_continueButton_tap() throws {
+        let button = try subject.inspect().find(button: Localizations.continue)
+        try button.tap()
+
+        waitFor(!processor.effects.isEmpty)
+
+        XCTAssertEqual(processor.effects.last, .startRegistration)
+    }
+
+    /// Tapping the region button dispatches the `.regionPressed` action.
+    func test_regionButton_tap() throws {
+        let button = try subject.inspect().find(
+            button: "\(Localizations.creatingOn): \(subject.store.state.region.baseUrlDescription)"
+        )
+        try button.tap()
+        waitFor(processor.dispatchedActions.last != nil)
+        XCTAssertEqual(processor.dispatchedActions.last, .regionTapped)
+    }
+
+    /// Tapping the receive marketing toggle dispatches the `.toggleReceiveMarketing()` action.
+    func test_receiveMarketingToggle_tap() throws {
+        if #available(iOS 16.0, macOS 13.0, tvOS 16.0, watchOS 9.0, *) {
+            throw XCTSkip("Unable to run test in iOS 16, keep an eye on ViewInspector to see if it gets updated.")
+        }
+        let toggle = try subject.inspect().find(viewWithId: ViewIdentifier.StartRegistration.receiveMarketing).toggle()
+        try toggle.tap()
+        XCTAssertEqual(processor.dispatchedActions.last, .toggleReceiveMarketing(true))
+    }
+
+    // MARK: Snapshots
```

Comment:
⛏️ Add another snapshot test for dark and extra large text with everything filled using long strings to see how multiline adapts in the view. No need to set dark and extra large on all tests.